### ISSUE TITLE
Request: Version of .alsoTo that takes a function

### ISSUE THREAD

**Issue by otto-dev** [akka/akka#28524]
`.wireTap` was added for execution of side-effects in a Flow without transforming the elements (see #23512). The problem is that it drops elements when the side-effect back-pressures.

You also have the `.alsoTo` operator. The difference is that `alsoTo` back-pressures instead of dropping the elements.

Furthermore, `.wireTap` can accept a either a `T => Unit` **or** a `Sink`. But `.alsoTo` only  accepts a `Sink`.

It would be nice to have a function-version of `alsoTo` like we have for `wireTap`, for situations where the side-effect is an essential part of the stream processing. An example would be telling an actor that an element has been processed:

```scala
  SomeSource()
      .ask(2)(actor)
      .via(???)
      .alsoTo(actor ! _) // tell actor the element is processed
      .alsoTo(println) // another dummy example
      .via(???) // do more with the element stream
```

If `alsoTo` is not a suitable name for this overload, my suggestion is `.sideEffect`.

Relevant PR: #24610 (`.wireTap`)

**Comment by johanandren** [comment]
I don't think we should add that.

For the actor use case, if that message interaction is not ask with a reply that arrives in the stream you won't know that it was delivered and it will not slow the stream down if the actor cannot keep up, but instead fill the mailbox and potentially the heap. I don't think fire-and-forget-without backpressure is something we should encourage even if it can make sense in some use cases.

If the operation should delay the stream element and you do not want to fan out for some reason I think `map { n => doStuff(n); n }`  or `mapAsync(1) { n => futureOp(n).map(n) }` isn't any extreme in syntax overhead vs how common I'd expect that need is. If it can happen in parallel the same goes for `.alsoTo(Sink.foreach(n => doStuff).async)`.

Note that if this is something you need very often in a specific project you should be able to add your own operators to the stream factories through implicit decoration/extension methods.

**Comment by otto-dev** [comment]
Thanks for getting back to me.

> you should be able to add your own operators to the stream factories through implicit decoration/extension methods

Thanks, I may do that.

Just for reference, the question of syntax and sensibility has been discussed along the same lines in #23512. My concern is that the resulting implementation drops elements on back-pressure, making it unsuitable for essential parts of the stream. I can see nothing in the original discussion leading up to this design choice, and I think it may well have sneaked under the radar, eg. due to the later choice of `.wireTap` for the operator name.

> it will not slow the stream down if the actor cannot keep up

Sending a message to an actor was an example use-case, but it could be any side-effect that does not need back-pressure. Having said that: In the above example back-pressure is already generated by the `.ask` on the same actor, preceding the `.sideEffect`.

Just adding my two cents. I'm coming from the same position as the original issue's author.

**Comment by huntc** [comment]
Having just come across this again, I also think that there's still utility in having a `foreach` style of stage. The main reason is to convey that a function is side-effecting.
